syncopy/io/load_tdt.py

```diff
@@ -0,0 +1,700 @@
+    out_name = tdt_info.info.blockname if out_path else None
+
+    tdt_data_handler = ESI_TDTdata(data_path, out_path, out_name,
+                                   subtract_median=False, channels=None, export=True)
+    adata = tdt_data_handler.data_aranging(file_paths, tdt_info)
+
+    return adata
+
+
+class ESI_TDTinfo():
+    def __init__(self, block_path):
+        self.block_path = block_path
+        self.nodata = False
+        self.t1 = 0
+        self.t2 = 0
+        self.UNKNOWN = int('00000000', 16)
+        self.STRON = int('00000101', 16)
+        self.STROFF = int('00000102', 16)
+        self.SCALAR = int('00000201', 16)
+        self.STREAM = int('00008101', 16)
+        self.SNIP = int('00008201', 16)
+        self.MARK = int('00008801', 16)
+        self.HASDATA = int('00008000', 16)
+        self.UCF = int('00000010', 16)
+        self.PHANTOM = int('00000020', 16)
+        self.MASK = int('0000FF0F', 16)
+        self.INVALID_MASK = int('FFFF0000', 16)
+        self.STARTBLOCK = int('0001', 16)
+        self.STOPBLOCK = int('0002', 16)
+        self.ALLOWED_FORMATS = [np.float32, np.int32, np.int16, np.int8, np.float64, np.int64]
+        self.ALLOWED_EVTYPES = ['all', 'epocs', 'snips', 'streams', 'scalars']
+
+    def code_to_type(self, code):
+        # given event code, return string 'epocs', 'snips', 'streams', or 'scalars'
+        strobe_types = [self.STRON, self.STROFF, self.MARK]
+        scalar_types = [self.SCALAR]
+        snip_types = [self.SNIP]
+
+        if code in strobe_types:
+            s = 'epocs'
+        elif code in snip_types:
+            s = 'snips'
+        elif code & self.MASK == self.STREAM:
+            s = 'streams'
+        elif code in scalar_types:
+            s = 'scalars'
+        else:
+            s = 'unknown'
+        return s
+
+    def time2sample(self, ts, fs = 195312.5, t1 = False, t2 = False, to_time = False):
+        sample = ts * fs
+        if t2:
+            # drop precision beyond 1e-9
+            exact = np.round(sample * 1e9) / 1e9
+            sample = np.floor(sample)
+            if exact == sample:
+                sample-= 1
+        else:
+            sample = np.ceil(sample) if t1 else np.round(sample)
+        sample = np.uint64(sample)
+        if to_time:
+            return np.float64(sample) / fs
+        return sample
+
+    def check_ucf(self, code):
+        # given event code, check if it has unique channel files
+        return code & self.UCF == self.UCF
+
+    def epoc_to_type(self, code):
+        # given epoc event code, return if it is 'onset' or 'offset' event
+        strobe_on_types = [self.STRON, self.MARK]
+        strobe_off_types = [self.STROFF]
+        if code in strobe_on_types:
+            return 'onset'
+        elif code in strobe_off_types:
+            return 'offset'
+        return 'unknown'
+
+    def code_to_name(self, code):
+        return int(code).to_bytes(4, byteorder = 'little').decode('cp437')
```

Comment:
wow.. for such a line a little one liner comment what is about to happen always helps fellow programmers ;)